samza-api/src/main/java/org/apache/samza/sql/schema/SqlFieldSchema.java

```diff
@@ -24,44 +24,52 @@
  */
 public class SqlFieldSchema {
 
-  private SamzaSqlFieldType fieldType;
-  private SqlFieldSchema elementType;
-  private SqlFieldSchema valueType;
-  private SqlSchema rowSchema;
+  private final SamzaSqlFieldType fieldType;
+  private final SqlFieldSchema elementType;
+  private final SqlFieldSchema valueType;
+  private final SqlSchema rowSchema;
+  private final Boolean isNullable;
+  private final Boolean hasDefaultValue;
 
-  private SqlFieldSchema(SamzaSqlFieldType fieldType, SqlFieldSchema elementType, SqlFieldSchema valueType, SqlSchema rowSchema) {
+  private SqlFieldSchema(SamzaSqlFieldType fieldType, SqlFieldSchema elementType, SqlFieldSchema valueType,
+      SqlSchema rowSchema, boolean isNullable, boolean hasDefaultValue) {
```

Comment:
Do you think it makes sense to call the hasDefaultValue field as optionalField? I think it also is a good idea to document  isNullable and isOptional in sqlFieldSchema to make the differences clear?